File: src/menu/components/TeamPanel.java
package menu.components;

import java.util.LinkedList;

import menu.Menu;
import menu.menues.LobbyMenu;
import menu.ComponentContainer;
import menu.components.Button;
import menu.components.Panel;
import menu.components.PlayerPanel;
import misc.game.Team;
import misc.math.menu.MenuRect;
import network.lobby.LobbyPlayer;;

public class TeamPanel extends Panel
{
	private Team team;
	private Button teamButton;

	public TeamPanel(ComponentContainer parent, MenuRect rect, Team team)
	{
		super(parent, rect);
		this.team = team; // Jedes TeamPanel representiert ein Team
		update(null);
	}

	public void update(LinkedList<LobbyPlayer> players)
	{
		boolean buttonWasEnabled;
		if (teamButton == null)
			buttonWasEnabled = true;
		else
			buttonWasEnabled = teamButton.isEnabled();

		getComponents().clear();
		getComponents().add(teamButton = new Button(this, new MenuRect(20, 20, 100, 30), "Team " + team.getName())
		{		
			@Override public void onClick(int mouseButton)
			{
				if (isEnabled())
					getLobbyMenu().teamPressed(team); // Übergabe an LobbyMenu, dass wir Team wechseln
			}					
		});

		teamButton.setEnabled(buttonWasEnabled);

		if (players == null)
		{
			return;
		}

		int i = 0;
		for (LobbyPlayer player : players)
		{
			if (player.getTeam().equals(team))
			{
				getComponents().add(new PlayerPanel(this, new MenuRect(getX() + 5 + i*65, 55, 60, 60), player));
				i++;
			}
		}

	}

	public void disableTeamButton() { teamButton.setEnabled(false); }

	private LobbyMenu getLobbyMenu() { return ((LobbyMenu) getParentMenu()); }
}

File: cpp/src/menu/components/TeamPanel.cpp
#include "TeamPanel.hpp"

#include "PlayerPanel.hpp"
#include <misc/Debug.hpp>
#include <menu/components/Button.hpp>
#include <menu/components/TeamListPanel.hpp>
#include <menu/menues/LobbyMenu.hpp>
#include <team/Team.hpp>
#include <player/LobbyPlayer.hpp>
#include <network/packets/TeamPacket.hpp>

TeamPanel::TeamPanel(LobbyMenu* m, TeamListPanel* c, const PixelRect& r, Team* team)
	: Panel(c, r), lobby(m), team(team)
{
	teamButton = NULL; // otherwise the program crashes
}

void TeamPanel::update(const std::vector<LobbyPlayer*>& players)
{
	bool buttonWasEnabled = (teamButton == NULL || teamButton->isEnabled());

	clearComponents();
	class TeamButton : public Button
	{
		public:
			TeamButton(TeamPanel* c, const PixelRect& r, const std::string& h) : Button(c, r, h) {}
			virtual void onPress() override
			{
				((TeamPanel*) getParent())->getLobbyMenu()->teamPressed(((TeamPanel*) getParent())->getTeam()); // Übergabe an LobbyMenu, dass wir Team wechseln
			}
	};
	addComponent(teamButton = new TeamButton(this, PixelRect(20, 20, 100, 30), "Team " + team->getName()));

	teamButton->setEnabled(buttonWasEnabled);

	int c = 0;
	for (unsigned int i = 0; i < players.size(); i++)
	{
		if (players[i]->getTeamPacket()->getTeam() == team)
		{
			addComponent(new PlayerPanel(players[i], getLobbyMenu(), this, PixelRect(getRelativeRect().getPosition().x + 5 + c*65, 55, 60, 60)));
			c++;
		}
	}
}

LobbyMenu* TeamPanel::getLobbyMenu() const
{
	return lobby;
}

Team* TeamPanel::getTeam() const
{
	return team;
}

void TeamPanel::disableTeamButton()
{
	teamButton->setEnabled(false);
}
